test/unit/spec/narrative_core/narrativeViewersSpec.js

```diff
@@ -119,14 +119,39 @@ define(['narrativeViewers', 'narrativeConfig'], (Viewers, Config) => {
             });
         });
 
+        it('should create a default viewer, given inputs and a missing widget', () => {
+            const dataCell = {
+                obj_info: {
+                    type: 'Module.Type1-1.0',
+                    bare_type: 'Module.Type1',
+                },
+            };
+            spyOn(window, 'require').and.callFake((files, cb, errCb) => {
+                errCb(new Error('Not a real widget!')); // make a fake error
+            });
+            return Viewers.createViewer(dataCell).then((view) => {
+                expect(window.require).toHaveBeenCalled();
+                expect(view.widget).toBeDefined();
+                expect(view.widget.prop('tagName')).toEqual('DIV');
+                expect(view.title).toEqual('Unknown Data Type');
+            });
+        });
+
         it('should create a viewer, given inputs', () => {
             const dataCell = {
                 obj_info: {
                     type: 'Module.Type1-1.0',
                     bare_type: 'Module.Type1',
                 },
             };
+            // spyOn(window, 'require').and.returnValue(new Error('nope'));
```

Comment:
comment accidentally left in?

Reply:
So much easier to remember these in the git diff viewer...